apps/src/studio/studio.js

```diff
@@ -4879,7 +4924,28 @@ Studio.askForInput = function (question, callback) {
 
     let handlerName = `askCallback${Studio.askCallbackIndex}`;
     Studio.askCallbackIndex++;
-    registerEventHandler(Studio.eventHandlers, handlerName, callback.bind(null, value || ''));
+    // Shift a CallExpression node on the stack that already has its func_,
+    // arguments, and other state populated:
```

Comment:
This is the same idea as [`codegen.createNativeInterpreterCallback`](https://github.com/code-dot-org/code-dot-org/blob/272f776281e472610d82495376a9b41337f050b7/apps/src/codegen.js#L397-L440) but it only executes until we get back to the starting depth then re-pauses the interpreter.

The plan is for this hack to go away once each callback runs in a separate JSInterpreter instance.